### PR TITLE
#4176 [Manager] Topics not added to the Pipeline should not appear in the edit properties

### PR DESCRIPTION
**Pull request by konekoya**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #4176)
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/Pipeline/PipelinePropertyDialog/PipelinePropertyDialog.js [hunk 1]
```diff
@@ -61,12 +61,33 @@ const PipelinePropertyDialog = props => {
...
+  // Display topics that are included in in the current pipeline
+  const getCurrentTopics = (pipeline, topics) => {
+    if (!pipeline || !topics) return [];
+    const validTopics = currentPipeline.objects
+      .filter(object => object.kind === KIND.topic)
+      .map(topic => topic.name);
+
+    const currentTopics = topics
+      .filter(topic => validTopics.includes(topic.name))
+      .sort((current, next) => {
```
Comment (konekoya) [inline ohara-manager/client/src/components/Pipeline/PipelinePropertyDialog/PipelinePropertyDialog.js]: Also sort the list

**Comment by konekoya** [comment]
retry manager-e2e

File: ohara-manager/client/src/components/Pipeline/PipelinePropertyDialog/PipelinePropertyDialog.js [hunk 2]
```diff
@@ -61,12 +61,33 @@ const PipelinePropertyDialog = props => {
...
   const { data: currentConnectors } = useConnectorState();
   const { data: currentStreams } = useStreamState();
-  const { data: currentTopics } = useTopicState();
+  const { data: allTopics } = useTopicState();
   const { data: currentFiles } = useFileState();
 
+  // Display topics that are included in in the current pipeline
+  const getCurrentTopics = (pipeline, topics) => {
+    if (!pipeline || !topics) return [];
+    const validTopics = currentPipeline.objects
```
Comment (eechih) [inline ohara-manager/client/src/components/Pipeline/PipelinePropertyDialog/PipelinePropertyDialog.js]: 既然方法有傳入 pipeline 變數，應該使用這個變數，降低耦合。

Comment (eechih) [reply ohara-manager/client/src/components/Pipeline/PipelinePropertyDialog/PipelinePropertyDialog.js]: 請問為何不直接使用 endpoints? objects 是 backend 根據 endpoints 去查詢的資料，我們無法確定資料的完整性。

Comment (konekoya) [reply ohara-manager/client/src/components/Pipeline/PipelinePropertyDialog/PipelinePropertyDialog.js]: > 既然方法有傳入 pipeline 變數，應該使用這個變數，降低耦合。

Oops

> 請問為何不直接使用 endpoints? objects 是 backend 根據 endpoints 去查詢的資料，我們無法確定資料的完整性。

Sounds legit to me :)

Comment (konekoya) [reply ohara-manager/client/src/components/Pipeline/PipelinePropertyDialog/PipelinePropertyDialog.js]: Comment addressed

**Comment by konekoya** [comment]
retry manager-api

**Comment by konekoya** [comment]
retry manager-ut

**Comment by konekoya** [comment]
retry manager-ut

**Comment by chia7712** [comment]
retry

**Comment by chia7712** [comment]
retry

**Comment by konekoya** [comment]
retry

**Comment by konekoya** [comment]
retry manager-e2e

**Pull request merged**